apps/designer/app/designer/features/style-panel/style-section.tsx

```diff
@@ -0,0 +1,777 @@
+          items={styleConfig.items}
+          // state={value.type === "invalid" ? "invalid" : undefined}
+          value={String(value.value)}
+          // @todo new combobox doesn't include any of these event handlers
+          // onValueSelect={setValue}
+          // onValueEnter={setValue}
+          // onItemEnter={(value: unknown) => {
+          //   setValue(value as string, { isEphemeral: true });
+          // }}
+          // onItemLeave={() => {
+          //   setValue(String(value.value), { isEphemeral: true });
+          // }}
+        />
+        <Unit value={value} />
+      </Flex>
+    </Grid>
+  );
+};
+
+const SelectControl = ({
+  currentStyle,
+  inheritedStyle,
+  setProperty,
+  styleConfig,
+}: ControlProps) => {
+  // @todo show which instance we inherited the value from
+  const value = getFinalValue({
+    currentStyle,
+    inheritedStyle,
+    property: styleConfig.property,
+  });
+
+  if (value === undefined) return null;
+
+  const setValue = setProperty(styleConfig.property);
+
+  return (
+    <Flex align="center">
+      <PropertyName
+        property={styleConfig.property}
+        label={styleConfig.label}
+        css={{
+          fontWeight: "500",
+          marginRight: "$sizes$1",
+        }}
+      />
+      <Select
+        options={styleConfig.items.map(({ label }) => label)}
+        value={String(value.value)}
+        onChange={setValue}
+        ghost
+        css={{
+          gap: "calc($sizes$1 / 2)",
+          px: "$sizes$1",
+          fontWeight: "500",
+          textTransform: "capitalize",
+          "&:hover": { background: "none" },
+        }}
+      />
+    </Flex>
+  );
+};
+
+const IconButtonWithMenuControl = ({
+  currentStyle,
+  inheritedStyle,
+  setProperty,
+  styleConfig,
+  category,
+}: ControlProps) => {
+  const value = getFinalValue({
+    currentStyle,
+    inheritedStyle,
+    property: styleConfig.property,
+  });
+  const isCurrentBreakpoint = useIsFromCurrentBreakpoint(styleConfig.property);
+
+  if (value === undefined) return null;
+
+  const setValue = setProperty(styleConfig.property);
+  const currentValue = value.value as string;
+
+  if (String(currentStyle.display?.value).includes("flex") !== true) {
+    return (
+      <ComboboxControl
+        currentStyle={currentStyle}
+        inheritedStyle={inheritedStyle}
+        setProperty={setProperty}
+        styleConfig={styleConfig}
+        category={category}
+      />
+    );
+  }
+
+  return (
+    <IconButtonWithMenu
+      label={styleConfig.label}
+      items={styleConfig.items}
+      value={String(currentValue)}
+      onChange={setValue}
+      icons={
+        (
+          icons as unknown as Record<
+            string,
+            Record<string, (props: unknown) => JSX.Element>
+          >
+        )[styleConfig.property]
+      }
+      css={{
+        ...(styleConfig.property !== "flexDirection" && {
+          svg: {
+            transform: `rotate(${
```

Comment:
I think you want to put the transform into the icon code, basically create the icon name you need and use the other icon + transform to fake a specific icon

Reply:
all those icon states seem like belong into the lib/IconButtonWithMenu, because IconButonWithMenuControl should only contain the glue code between the render component and the state